sources/api/netdog/src/cli/mod.rs

```diff
@@ -92,7 +104,11 @@ fn primary_interface_name() -> Result<String> {
 
 /// Potential errors during netdog execution
 mod error {
-    use crate::{dns, interface_id, lease, net_config, wicked};
+    #[cfg(net_backend = "wicked")]
+    use crate::lease;
+    #[cfg(net_backend = "systemd-networkd")]
+    use crate::networkd_status;
+    use crate::{dns, interface_id, net_config, wicked};
```

Comment:
Why not this? It doesn't compile correctly - is that just because "full" networkd support is waiting on more of this logic to be conditionalized?

```suggestion
    #[cfg(net_backend = "wicked")]
    use crate::{lease, wicked};
    #[cfg(net_backend = "systemd-networkd")]
    use crate::networkd_status;
    use crate::{dns, interface_id, net_config};
```

Reply:
Correct, the logic for `generate_net_config.rs` hasn't been touched for the conditional compilation and a future PR should address that. It starts to bleed into doing that work if this is pulled out of the systemd-networkd side. I can go in and do that but wanted to avoid stepping on that parallel work if possible. Right now this does result in a not-yet-functional network if you are using systemd-networkd but this PR isn't making this problem worse, its also not intended to solve everything. I'm happy to strike a different balance on where the conditional compilation is cut if we need to though.

Reply:
I'll be handling the rest of this with the changes to generate networkd config - with this PR let's get the program to compile and not affect any existing variants.